OIPA/iati/parser/tests/test_2_03_parser.py

```diff
@@ -2926,6 +2926,138 @@ def test_activity_result_indicator_baseline(self):
+
+        self.parser_203.register_model('Activity', self.activity)
+
+        self.parser_203.register_model(
+            'ResultIndicatorBaseline',
+            self.result_indicator_baseline
+        )
+        self.parser_203.register_model('Location', self.location)
+
+    def test_activity_result_indicator_baseline_location(self):
+
+        """
+        Test if <location> element in context of a baseline element (as part
+        of a parent result/indicator element) is parsed and saved correctly
+        """
+
+        # 1. 'ref' attribute is not provided:
+        result_indicator_baseline_location_attrs = {
+            # 'ref': 'AF-KAN',
+        }
+
+        result_indicator_baseline_location_XML_element = E(
+            'location',
+            **result_indicator_baseline_location_attrs
+        )
+
+        try:
+            self.parser_203\
+                .iati_activities__iati_activity__result__indicator__baseline__location(  # NOQA: E501
+                    result_indicator_baseline_location_XML_element
+                )
+        except RequiredFieldError as inst:
+            self.assertEqual(inst.field, 'ref')
+            self.assertEqual(inst.message,
+                             "This attribute has to be a cross-reference to "
+                             "the internal reference assigned to a defined "
+                             "location: iati-activity/location/@ref, so "
+                             "leaving it blank makes no sense")
+
+        # 2. Referenced location doesn't exist:
+        result_indicator_baseline_location_attrs = {
+            'ref': 'AF-KAN',  # it's not 'KH-PNH'
+        }
+
+        result_indicator_baseline_location_XML_element = E(
+            'location',
+            **result_indicator_baseline_location_attrs
+        )
+
+        try:
+            self.parser_203\
+                .iati_activities__iati_activity__result__indicator__baseline__location(  # NOQA: E501
+                    result_indicator_baseline_location_XML_element
+                )
+        except FieldValidationError as inst:
+            self.assertEqual(inst.field, 'ref')
+
+            self.assertEqual(
+                inst.message,
+                "Referenced location doesn't exist"
+            )
+
+        # 3. Referrenced location isn't refferenced on an activity level:
+        result_indicator_baseline_location_attrs = {
+            'ref': 'KH-PNH',
+        }
+
+        result_indicator_baseline_location_XML_element = E(
+            'location',
+            **result_indicator_baseline_location_attrs
+        )
+
+        try:
+            self.parser_203\
+                .iati_activities__iati_activity__result__indicator__baseline__location(  # NOQA: E501
+                    result_indicator_baseline_location_XML_element
+                )
+        except FieldValidationError as inst:
+            self.assertEqual(inst.field, 'ref')
+
+            self.assertEqual(
```

Comment:
Doesn't test this condition. We probably need to assigned different 'ref' in activity element so that the 'FieldValidationError' exception is triggered.

Reply:
Umm but the code block does exactly that :D It checks if `FieldValidationError ` is raised if `Location` doesn't exist in `self.activity.location_set.all()`. Later on int he 4rd case (when all is good) it adds it to the Activity locations to make sure this error is not raised :) See: https://github.com/zimmerman-zimmerman/OIPA/pull/824/files#diff-b9b3dce80d795ec2171b5737b9132331R3046